vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -124,6 +124,13 @@ def __init__(self, name, init_config, agentConfig, instances):
         self.metrics_metadata = {}
         self.latest_event_query = {}
 
+        self.hostname_upper = self.hostname.upper()
+
+    def sanitize_hostname(self, hostname):
```

Comment:
does this only happen on the box that the vsphere check is running from or does it happen from any box that vsphere manages that's running an agent?

Reply:
My understanding is that it's the former.

Reply:
It's weird that only the one hostname would be duplicated. Why is the hostname the check reports affected by it being the box the check is running on?

Reply:
I assume the hosts vsphere returns includes the one where the check is running. Given that, when we provide the hostname from vsphere here:

https://github.com/DataDog/integrations-core/blob/1fe3a88915b646336adf66cc4cebde0f774c1841/vsphere/datadog_checks/vsphere/vsphere.py#L663

the hostname may be different than the one we detect (when we don't supply the hostname for other metric submission calls):

https://github.com/DataDog/integrations-core/blob/1fe3a88915b646336adf66cc4cebde0f774c1841/vsphere/datadog_checks/vsphere/vsphere.py#L695

Reply:
But the agent can be running on the other hosts as well, just not running the check. Will this fix the mismatch for those?

Reply:
(If there is a mismatch on those)

Reply:
Yes because the issue arises only when the running agent tries to detect the hostname. The other hostnames would be coming directly from vsphere.